### ISSUE TITLE
Restore compatibility with LiberTEM v0.9.0.dev0

### ISSUE THREAD

**Issue by sk1p** [LiberTEM/LiberTEM-live#37]
Currently errs out when dry-running with the following error:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_1989238/3821036961.py in <module>
      4 # LivePlot = MPLLive2DPlot
      5 
----> 6 p0 = LivePlot(aq, udfs[0])
      7 p1 = LivePlot(aq, udfs[1])
      8 p2 = LivePlot(aq, udfs[2])

~/source/LiberTEM/src/libertem/viz/bqp.py in __init__(self, dataset, udf, roi, channel, title, min_delta, udfresult)
     57             self, dataset, udf, roi=None, channel=None, title=None, min_delta=1/60, udfresult=None
     58     ):
---> 59         super().__init__(
     60             dataset=dataset,
     61             udf=udf,

~/source/LiberTEM/src/libertem/viz/base.py in __init__(self, dataset, udf, roi, channel, title, min_delta, udfresult)
    155     ):
    156         if udfresult is None:
--> 157             udfresult = UDFRunner.dry_run([udf], dataset, roi)
    158         eligible_channels = get_plottable_2D_channels(udfresult.buffers[0])
    159         if channel is None:

~/source/LiberTEM/src/libertem/udf/base.py in dry_run(cls, udfs, dataset, roi)
   1442         runner = UDFRunner(udfs)
   1443         executor = InlineJobExecutor()
-> 1444         res = runner.run_for_dataset(
   1445             dataset=dataset,
   1446             executor=executor,

~/source/LiberTEM/src/libertem/udf/base.py in run_for_dataset(self, dataset, executor, roi, progress, corrections, backends, dry)
   1740         dry: bool = False
   1741     ) -> "UDFResults":
-> 1742         for res in self.run_for_dataset_sync(
   1743             dataset=dataset,
   1744             executor=executor.ensure_sync(),

~/source/LiberTEM/src/libertem/udf/base.py in run_for_dataset_sync(self, dataset, executor, roi, progress, corrections, backends, dry)
   1762         dry: bool = False
   1763     ) -> Generator["UDFResults", None, None]:
-> 1764         tasks, params = self._prepare_run_for_dataset(
   1765             dataset, executor, roi, corrections, backends, dry
   1766         )

~/source/LiberTEM/src/libertem/udf/base.py in _prepare_run_for_dataset(self, dataset, executor, roi, corrections, backends, dry)
   1709         # Other boundary conditions when moving input data to device
   1710         # FIXME: approximate partition shape here
-> 1711         partition = next(dataset.get_partitions())
   1712         tiling_scheme = neg.get_scheme(
   1713             udfs=self._udfs,

~/source/modules/LiberTEM-live/src/libertem_live/detectors/merlin/acquisition.py in get_partitions(self)
    131 
    132         header = self.source.socket.get_acquisition_header()
--> 133         frame_header = self.source.socket.get_first_frame_header()
    134 
    135         self.source.validate_get_sig_shape(frame_header, self._sig_shape)

~/source/modules/LiberTEM-live/src/libertem_live/detectors/merlin/data.py in get_first_frame_header(self)
    467 
    468     def get_first_frame_header(self):
--> 469         return self._first_frame_header
    470 
    471     def read_headers(self, cancel_timeout=None):

AttributeError: 'MerlinDataSocket' object has no attribute '_first_frame_header'
```

Aside: maybe add a CI job to run LiberTEM-live against the master LiberTEM version, in at least one Python version, to increase the chance we catch these errors before it's release time.

**Comment by uellue** [comment]
> Aside: maybe add a CI job to run LiberTEM-live against the master LiberTEM version, in at least one Python version, to increase the chance we catch these errors before it's release time.

👍 👍 👍

**Comment by sk1p** [comment]
Fixed by #45 - I'll work on improving the CI by adding an integration job to the main LiberTEM repo.
